### PR TITLE
fixes #126, interpolated variables as property names

### LINKED ISSUE #126: LESS Mixin Variable Rule Throwing Parse Error
Mixin variable rules are throwing parse errors in 3.2.1. Sample less mixin below:

```
.browser-prefix(@prop, @args) {
  -webkit-@{prop}: @args;
  -moz-@{prop}: @args;
  -ms-@{prop}: @args;
  -o-@{prop}: @args;
  @{prop}: @args;
}
```

Parse error is on the last line in the mixin: `@{prop}: @args;`

### PR DESCRIPTION
**Pull request by shellscape**
Fixes interpolated variables as property names in mixins

### CODE REVIEW

File: test/less/declaration/2.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
+  "type": "declaration",
+  "content": [
+    {
+      "type": "property",
+      "content": [
+        {
+          "type": "ident",
+          "content": "@{prop}",
```
Comment (tonyganch) [inline test/less/declaration/2.json:9]: `@{prop}` is not an ident, it is a variable, so it should be parsed as a variable. I'll take a look at this one later.

Comment (shellscape) [inline test/less/declaration/2.json:9]: I don't understand why not. That's how the parser is setup to handle that part. This fixes a pretty critical oversight and **breaking bug** in the 3.x versions that worked fine with 3.0.0.-31. At the very least, with all tests passing, it should be merged and "corrected" in a patch version.

Comment (tonyganch) [inline test/less/declaration/2.json:9]: `ident` is something different. Here is grammar from the spec:
http://www.w3.org/TR/css3-selectors/#lex

And according to Less's spec, `@{prop}` is an `interpolatedVariable`:
http://lesscss.org/features/#variables-feature-variable-interpolation
https://github.com/tonyganch/gonzales-pe/blob/dev/docs/node-types.md#interpolatedvariable

**Comment by shellscape** [comment]
My point is that 3.x of your module is throwing exceptions on valid LESS syntax. If you choose not to use this PR (or the others) that's perfectly reasonable, but it's your call to continue to allow your module to break on valid syntax. While my fix isn't how you would have gone about it, it's a good starting point, and it does more than the module currently does. When I get a PR from someone on my projects that fixes an overall issue, but doesn't work exactly how I would've done it, I accept the PR and modify to my preferences (or how I think a spec should be implemented) in order to advance the project.

Completely your call, but I think it'd be prudent to follow that same mantra. Unfortunately I don't have any more time to give towards the existing PRs.

**Comment by sascha-egerer** [comment]
Any progress?

**Comment by shellscape** [comment]
@sascha-egerer it doesn't seem like @tonyganch has any plans for fixing everything that the last version broke with regard to LESS. I've moved on to use PostCSS and https://github.com/webschik/postcss-less. it's faster, easier to work on and with.

**Comment by sascha-egerer** [comment]
@shellscape thanks for the hint but this packages is just a dependeny of another package i'm using. So no way for me to switch for now...

**Pull request closed without merging**